src/Features/LanguageServer/Protocol/Handler/ProjectContext/GetTextDocumentWithContextHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [Shared]
+    [ExportLspMethod(MSLSPMethods.ProjectContextsName)]
+    internal class GetTextDocumentWithContextHandler : IRequestHandler<GetTextDocumentWithContextParams, ActiveProjectContexts?>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public GetTextDocumentWithContextHandler()
+        {
+        }
+
+        public Task<ActiveProjectContexts?> HandleRequestAsync(
+            Solution solution,
+            GetTextDocumentWithContextParams request,
+            ClientCapabilities clientCapabilities,
+            string? clientName,
+            CancellationToken cancellationToken)
+        {
+            var documents = solution.GetDocuments(request.TextDocument.Uri, clientName);
+
+            if (!documents.Any())
+            {
+                // TODO: am I supposed to return an empty list of contexts or null here?
+                return Task.FromResult<ActiveProjectContexts?>(null);
+            }
+
+            var contexts = new List<ProjectContext>();
+
+            foreach (var document in documents)
+            {
+                var project = document.Project;
+
+                contexts.Add(new ProjectContext
+                {
+                    Id = project.Id.Id.ToString(),
+                    Kind = project.Language == LanguageNames.CSharp ? ProjectContextKind.CSharp : ProjectContextKind.VisualBasic,
```

Comment:
would it be good to explicitly check VB rather than set everything not csharp to VB?

Reply:
ProjectContextKind is an enum, so I had assumed I had to return _something_ but I guess it has no value defined for zero. :-/